client/src/components/App.jsx

```diff
@@ -37,23 +35,28 @@ class App extends React.Component {
     });
   }
 
+  getId() {
+    const id = Number(window.location.pathname.split('/')[1]);
+    console.log('id', id);
+    return id;
+  }
 
   componentDidMount() {
-    this.getInfo();
+    console.log('appl', App)
+    const id = this.getId();
```

Comment:
You may want to not have `getId` as its own function, as this is technically running async and may not complete in time for `getInfo` to run. Putting line 39 here and just setting the variable that way may be the way to go.

Reply:
Yeah, that makes sense. I had originally followed the separation of concerns methodology to make it easier to mock this function for a test, which unfortunately, proved unsuccessful.